src/operators/fully-connected-nc.c

```diff
@@ -1780,31 +1922,58 @@ static enum xnn_status reshape_fully_connected_nc(
     if (xnn_is_hmp_gemm_ukernel(gemm_ukernel)) {
       fully_connected_op->compute[0].type = xnn_parallelization_type_2d_tile_2d_with_uarch;
       if (dynamic_quantization) {
-        fully_connected_op->compute[0].task_2d_tile_2d_with_id = (pthreadpool_task_2d_tile_2d_with_id_t) xnn_compute_hmp_dqgemm;
+        if (is_blockwise_kernel) {
+          fully_connected_op->compute[0].task_2d_tile_2d_with_id = (pthreadpool_task_2d_tile_2d_with_id_t) xnn_compute_hmp_dqgemm_bl;
+        } else {
+          fully_connected_op->compute[0].task_2d_tile_2d_with_id = (pthreadpool_task_2d_tile_2d_with_id_t) xnn_compute_hmp_dqgemm;
+        }
```

Comment:
Actually, I would remove the `xnn_compute_hmp_dqgemm_bl` function and just use `xnn_compute_hmp_dqgemm` to avoid this special-casing here and below.